### DISCUSSION TITLE
Using Z2M during developing and testing custom zigbee devices

### DISCUSSION THREAD

**Discussion by phalox** [Koenkk/zigbee2mqtt#19533] [General]
Hi @Koenkk, bedankt voor je fantastische werk om zigbee devices zo accessible te maken in home assistant!

I'm experimenting with creating some new Zigbee devices based on CC2530 boards. To avoid writing firmware, I'm using the [firmware builder of ptvo](https://ptvo.info/). This worked so far, but as soon as you want to add an output it becomes a hassle to 'refresh' the device definition in Z2M. The developer of this project gave me these instructions:

https://github.com/ptvoinfo/zigbee-configurable-firmware/issues/226#issuecomment-1704570529

I understand that Z2M's focus is not for developers in the first place, but rather for finished products. But while developing, I very quickly feel the need to test things out end2end in home assistant, and therefore need to pass by Z2M. Any suggestions how this flow could be improved?

Thanks!

**Comment by Koenkk** [comment]
One thing you could do is to write a [custom user extension](https://www.zigbee2mqtt.io/advanced/more/user_extensions.html) which calls the [`interview()`](https://github.com/Koenkk/zigbee-herdsman/blob/e54db7a20bb442d8be7bef2de837772cdcbfaec3/src/controller/model/device.ts#L476) on the device.
